### PR TITLE
Pptp 2134 - Submission schema validations

### PR DESCRIPTION
**Pull request by JoeSimmonds**
### Description of Work carried through
Validates the Submissions against a schema and logs any errors.

Some refactoring also:
1. When I went to add the logging the controller had all the logic, and I didn’t want to just add more to that ball of mud.  So I’ve added a new SubscriptionService.  It has the same logic but has no interest in the HTTP level stuff.
2. The registration request had an Option[Map] in it which was unnecessary, and could just be a Map.
3. The logging contained lots of identifiers for orgs and I’ve obfuscated those with MD5 hashes.  We should now be able to tell when these things are the same but be able to know which ones they are without prior knowledge.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] Required Environment Config has been amended/added
 - [x] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+object Obfuscation {
+  private val logger = LoggerFactory.getLogger("application." + getClass.getCanonicalName)
+
+  def obfuscate(value: String): String = {
+    val result = MessageDigest.getInstance("MD5")
+      .digest(value.getBytes(StandardCharsets.UTF_8))
+      .map(c => "%02X".format(c)).mkString
+
+    logger.debug(s"obfuscated [$value] as [$result]")
```
Comment (lmcewanhmrc) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala:32]: Did you mean to leave this in?

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/services/SchemaValidatorService.scala [hunk 2]
```diff
@@ -14,18 +14,8 @@
...
-  statusCode: Int
-) extends SubscriptionResponse
-
-object SubscriptionFailureResponseWithStatusCode {
-
-  implicit val format: OFormat[SubscriptionFailureResponseWithStatusCode] =
-    Json.format[SubscriptionFailureResponseWithStatusCode]
-
-}
+class SchemaValidatorService @Inject() {}
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/services/SchemaValidatorService.scala:21]: do we need this?

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    result
+  }
+
+  implicit class StringObfuscationOps(inner: String) {
```
Comment (liampan) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala:36]:
```suggestion
  implicit class StringObfuscationOps(val inner: String) extends AnyVal {
```

**Review by lmcewanhmrc — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala (hunk 1 above)
Comment (JoeSimmonds) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala:32]: I did, but happy to take it out.  

It means when you have debug logging enabled you can see what each hash value actually is.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/services/SchemaValidatorService.scala (hunk 2 above)
Comment (JoeSimmonds) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/services/SchemaValidatorService.scala:21]: No, I'll remove that.  It was my first thought about how to implement

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala (hunk 1 above)
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/util/Obfuscation.scala:32]: Up to you - just checking 😸

**Review by ajchilts — APPROVED** [review]

**Pull request merged**